bin/live/pycbc_live_collate_triggers

```diff
@@ -0,0 +1,280 @@
+                # The stored chisq is actually reduced chisq, so convert back
+                # to unreduced chisq using chisq_dof
+                tmpchisq = triggers['chisq'][:] * (2 * triggers['chisq_dof'][:] - 2)
+                triggers['chisq'][:] = tmpchisq
+
+                # Apply the cuts to triggers
+                keep_idx = cuts.apply_trigger_cuts(triggers, trigger_cut_dict)
+
+                # triggers contains the datasets that we want to use for
+                # the template cuts, so here it can be used as the template bank
+                keep_idx = cuts.apply_template_cuts(
+                    triggers,
+                    template_cut_dict,
+                    template_ids=keep_idx
+                )
+                if not any(keep_idx):
+                    # No triggers kept after cuts in this ifo for this file
+                    continue
+                n_triggers_cut[ifo] += keep_idx.size
+
+                triggers = {
+                    k: triggers[k][keep_idx]
+                    for k in triggers.keys()
+                }
+
+                if ('approximant' not in triggers) or (len(triggers['approximant']) == 0):
+                    continue
+
+                for name, dataset in triggers.items():
+                    if name in destination[ifo]:
+                        # Append new data to existing dataset in destination
+                        if triggers[name].shape[0] == 0:
+                            continue
+                        destination[ifo][name].resize(
+                            n_triggers_cut[ifo],
+                            axis=0
+                        )
+                        destination[ifo][name][-keep_idx.size:] = triggers[name]
+                    else:
+                        destination[ifo].create_dataset(
+                            name,
+                            data=dataset,
+                            chunks=True,
+                            maxshape=(None,)
+                        )
+
+                for attr_name, attr_value in source.attrs.items():
+                    destination.attrs[attr_name] = attr_value
+
+    for ifo in args.ifos:
+        # Collect the segments, and output to the file
+        search_grp = destination[ifo].create_group('search')
+        segs[ifo].coalesce()
+        seg_starts, seg_ends = veto.segments_to_start_end(segs[ifo])
+        search_grp.create_dataset(
+            'end_time',
+            data=seg_ends,
+        )
+        search_grp.create_dataset(
+            'start_time',
+            data=seg_starts,
+        )
+        logging.info(
+            "Found %d %s triggers in %s seconds",
+            n_triggers[ifo],
+            ifo,
+            abs(segs[ifo])
+        )
+        if n_triggers[ifo] != n_triggers_cut[ifo]:
+            logging.info(
+                "  %d triggers after cuts",
+                n_triggers_cut[ifo],
+            )
+
+with h5py.File(args.bank_file,'r') as bank_file:
```

Comment:
Should this be moved to the start, then you can avoid closing/reopening the output file?

Reply:
Looks like it should